### DISCUSSION TITLE
You may regret the Mit/Apache 2.0 license stack.

### DISCUSSION THREAD

**Discussion by Destroyinator69420** [freenet/freenet-core#318] [General]
I am writing about a concern that you may regret your permissive license stack. Many open source projects have used permissive licenses, only to regret it when a faceless corporation forks the source code to make a freedom subtracted software package without giving anything back to the people that originally developedthe software. What happens when Amazon turns Locutus into a product, makes enhancements, and then refuses to give anything back to you, not even the documentation on how to implement the changes yourself. This has happened to many projects, such as netbsd, which apple and microsoft forked the network stack from without giving anything back to netBSD, not even a statement of gratitude. Amazon Web Services has taken many opens source projects without giving back to the people who made them. I do not want this to happen to you. Rememer that once this happens to you, you have no right to complain, since you allowed them to do it.  I would recommend you upgrade to a weak copyleft license to preserve the freedom of others. I would recommend the LGPL 3.0 for the citcumstances of Locutus, since you want to allow code written for locutus to run under its own license. This will also allow Locutus to be combined into Web Browsers and other apps while preserving Freedom of those who use Locutus. This will make sure that corporations that use Locutus still have to give back to the community. The best time to make this change is after Locutus 1.0 releases to the General Public.

**Comment by sanity** [comment]
The choice of license was a condition of a funding grant that paid for quite a bit of development over the past year.

I agree with you that this has been a problem with open source. However, it's difficult for me to imagine Amazon being able to turn Locutus into a product.

That said, we now have other funding sources that don't have that restriction, so some components will likely be LGPL going forward (where we think a more Stallmanesque license is appropriate).

**Reply by Destroyinator69420** [reply]
Now that I think of it, the LGPL is not good enough, since it would allow cancer nodes to infect the network while being under no obligation to release their source code. This is a problem with Freenet classic where the levine group spys on freenet and makes false criminal accusations with impunity since they are under no obligation to release their source code, so Arnebab and tje freenet classic devs cannot fix the exploit they are using. If we want to protect Locutus from cancer nodes, we need to be able go sue cancer node operatora for not releasing their source code. The only licenses that can fix this are the GNU Affero GPL v3 or later or the Open Software License 3.0. I recommend the GNU Affero GPL v3 or later since it will prevent people from enforcing the DMCA anti circumvention law over Locutus and protects against patent cartels. It will also prevent jeff bezos or others from turning Locutus proprietary by keeping it internal to the organization while interacting with the rest of the network. The Affero GPL v3 or later is also necessary becuase it will allow locutus to upgrade to newer versions of the GNU Affero GPL to address new concerns and threats to free software. It is the only way to keep locutus free. We can use a license the WASM runtime and STDLIB under the LGPL-3.0 or later so that smart contracts can be under a different license, by the Locutus core needs stronger copyleft to protect the freedom of the people using it.

**Reply by sanity** [reply]
Yes, I've given the AGPL some thought. The challenge is that an excessively restrictive license like the AGPL may be a disincentive for future grants. I'm still ruminating on it, we have time to make a final decision.

**Comment by aschrijver** [comment]
Tangential: Do you intend to be transparent on who these funding providers are? Or is that info already available somewhere?

**Reply by sanity** [reply]
Typically, US 501c3 non-profits aren't required to reveal their donors to the public, and there are some valid reasons that we wouldn't want to.

That said, our latest grants came from [FUTO](https://futo.org/) and [Protocol Labs](https://protocol.ai/), both announced at the time.

**Reply by aschrijver** [reply]
Thank you!
